apsi-enrollment-fe/src/app/modules/main/pages/event-detail/event-detail.component.ts

```diff
@@ -39,22 +40,28 @@ export class EventDetailComponent implements OnInit {
   }
 
   onRegister() {
-    this.eventService
-      .signUp(this.event.id)
-      .pipe(
-        tap((enrollment) => {
-          this.enrollmentStatus = enrollment.status;
-        }),
-        switchMap((enrollment) => this.paymentService.create(enrollment))
-      )
-      .subscribe(
-        (payment) => {
-          window.location.href = payment.redirectUrl;
-        },
-        (error) => {
-          console.error(error);
-          // TODO: use API error to display error message alert
-        }
-      );
+    this.createPayment(this.eventService.signUp(this.event.id));
+  }
+
+  onPayNow() {
+    this.createPayment(this.eventService.getEnrollment(this.event.id));
+  }
+
+  createPayment(enrollment: Observable<Enrollment>) {
```

Comment:
In most cases observable should not be argument of a function. You should use `switchMap`.

Reply:
changed